@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -33,6 +34,11 @@ export class TextElementState {
      * Used during rendering.
      */
     private m_textRenderState?: RenderState;
+    /**
+     * @hidden
+     * Used to store recently used text layout.
+     */
+    private m_textLayoutState?: LayoutState;
```

Comment:
A performance issue that we have in labeling is the allocation of several small objects for each label handled by the renderer. That leads to thousands of small allocations. This here adds yet another object, maybe we could combine it with RenderState or flatten the structure and put everything in TextElementState?

Reply:
How does it comply with you previous statement -> define - construct LayoutState in TextElementState c-tor?
We need to decide on something:
- either always allocated in c-tor - simplifies the code (also helps JIT),
- always allocated when TextElementState initialized - current code,
- allocated for features using it - for all point labels, requires some more code branches to current code,
- never allocated, but only on demand (point labels when placed alternatively) -> most difficult to hassle in the logic,